# Exploring-the-Microbiome-

# Team participant 
  - Casey Eddington - Graduate student at University of Minnesota 
  - Islam EL JADDAOUI -PhD student at University Mohammed V 
  - Sofia Sehli- PhD student at University Mohammed VI of Health Sciences
  - Zainab El Ouafi-  PhD student at University Mohammed VI of Health Sciences 
  - Soumaya Jbara - PhD student at University Mohammed V
  - Arthur Shem Kasambula - MSc Student Makarere University Uganda 
  
 # Introduction
   Since the beginning of covid-19 epidemic, one of the most prevalent symptoms observed in infected individuals have been gastrointestinal issues such as diarrhea, stomach pains, and vomiting (Villapol, 2020). It occurred to many researchers to investigate the microbiota diversity in infected patients, and determine the association between the gut microbiota and the severity of the covid-19 infection as well as the human immune response.In this research, we introduce [name] pipeline, which was created to reduce time and effort while eliminating manual mistakes, especially when dealing with large amounts of data. This metagenomics analysis process includes assembly and annotation, allowing for rapid production and accurate data delivery. (See Figure 1)
  
 # Workflow 
 
![image](https://user-images.githubusercontent.com/85350037/120885511-bb93a780-c5e9-11eb-9674-38dc9f43d37b.png)

# List of Tools
WGS
- FastQC: 
- Metaspades: 
- Prodigal:
- Mgrast:
- Metaphlan: 

16S
- DADA2:
- Phyloseq:
- ggplot2:
- 


