# TAXAPRO

### What is the problem?
The ability to promptly sequence whole genomes at a relatively low cost has revolutionized the way we study microbiomes. Microbiologists are no longer limited to studying what can be grown in a lab and instead are given the opportunity to rapidly identify the makeup of microbial communities in a wide variety of environments. Analyzing whole genome sequencing (WGS) data is a complex process that involves multiple moving parts and might be rather unintuitive for scientists that donâ€™t typically work with this type of data. Thus, to help lower the barrier for less-computationally inclined individuals, TAXAPRO was developed at the 2021 African Society for Bioinformatics and Computational Biology (ASBCB) Omics Codeathon. TAXAPRO is an advanced metagenomics pipeline that accurately assembles organelle genomes from whole genome sequencing data. TAXAPRO seamlessly combines WGS analysis tools to create a pipeline that automatically processes raw WGS data and presents organism abundance information in both a tabular and graphical format.  


### Why should we solve it?
While metagenomics has evolved as a preferred tool for examining bacterial populations, assembling metagenomic data remains difficult, impeding biological discoveries. Furthermore, increasing the number of proper gene identifications and translation start locations for each gene, as well as lowering the overall number of false positives, are all desired outcomes. To address this challenges the TAXAPRO pipeline designed to automatically analyze large WGS datasets reducing human intervention, potential for error, and the possible computation barrier involved in such a task and delivers accurate strain-level profiling, with a taxonomic classifications that are unambiguous; accurate calculation of organismal relative abundance; species-level resolution; and strain identification and tracking.


### What is TAXAPRO?
TAXAPRO is an automated analytic pipeline developed using python at the ASBCB Omics Codeathon in June 2021. TAXAPRO is a microbiological profiling script code that delivers accurate strain-level profiling. Taxonomic classifications that are unambiguous; accurate calculation of organismal relative abundance; species-level resolution; and strain identification and tracking. It allows us to determine the taxonomic diversity and the functional aspects of the gut microbiome of CoVID-19 patients in our case.

### Overall Workflow
Here is the workflow for the pipeline:
![image](https://user-images.githubusercontent.com/85350037/121264719-e61a8480-c8b7-11eb-8846-4e0062df924c.png)

### How to use TAXAPRO
The most important point in the usage of TAXAPRO is that the scripts must be in the same location as all the **paired-end** Fastqs (*the reads must be paired-end sequences*) that you're looking to analyse.
Then you can cd into the Fastq directory and then use python to run the scripts from there:

```
cd <location_of_fastqs>
python TaxPro.py
```

Installation options:
#### Pipenv
1- Install pipenv
```
pip install pipenv
```
2- Activate the environment
```
cd <location_of_the_script>
pipenv shell
```

#### Docker
We will provide a docker image for installing and using SARS2MicrobeTaxa.

The Docker image will contain SARS2MicrobeTaxa and it will be downloadable from the Docker Hub.
We will be able to run this tool as a Docker or Singularity container.

- docker pull <!-- omicscodeathon/microbetaxasars2 -->: Command to pull the image from the DockerHub

- docker run <!-- omicscodeathon/microbetaxasars2 -->: Run the docker image from the command line


### Methods

In this project, we analysed Whole Genome Sequences (WGS) from 6 CoVID-19 patients and the dataset used was obtained from the Sequence Read Achive (SRA). These were fecal samples from CoVID-19 patients.
TheTAXAPRO pipeline implements the following steps:
- [x] TAXAPRO uses SRAdownload to pull raw WGS data from the SRA database. It uses FastQC to provide an overview of basic quality control for NGS sequencing data by exporting an HTML file that flags the FASTQ sequences as passed, warn, or fail. 
- [x] TAXAPRO then uses Trimmomatic to clean the data. 
- [x] Trimmomatic passes the cleaned data to MetaSPAdes for genome assembly. 
- [x] Prodigal uses an unsupervised machine learning algorithm to annotate the assembled genome providing nucleotide and amino acid sequence data belonging to each species. 
- [x] Sequences are then mapped against the non-redundant database using blast and blastp to provide functional assignment.
- [x] Results are compiled and presented visually using the python package seaborn. 


![Copie de Organizational Charts by Slidesgo](https://user-images.githubusercontent.com/85350037/139056048-f035cc30-d5e6-4c2f-b7ed-5e1e30f44384.png)


### Results
Results of the analysis are in the [output folder](output/).

The evaluation of TAXAPRO pipeline showed a improuvement in the analysis time span, tendency to make mistakes, and computational intensivity. The results showed significant assembly differences between species, and a smooth mapping plot. 
There were no significant variations in the composition of the gut microbiome of the 6 cocid-19 samples of infected patients. According to our findings and the use of the TAXAPRO pipelines, Clostridia and Bacteroidia genuses were discovered in larger quantities in the fecal bacterial samples. The microbiota profiles of the six samples were rather similar. We found a rise in the relative abundance of _Bacteroides fragilis, Desulfotomaculum acetoxidans, Pelotomaculum thermopropionicum, Escherichia coli_, and other opportunistic bacterial strains at the strain level.
These findings add to our understanding of the alteration of the gut microbiome in covid-19 infected individuals, paving the way for a better understanding of the microbiota's effects on people's health.


### Conclusion and perspective
TAXAPRO was developed to make WGS analysis easier and more accessible for all. By simply running a script overnight, the user will have access to an easy to interpret organism abundance analysis that was shown to be accurate using data from BioProject 624223. Considering the automation, even WGS-experts might want to use a pipeline like TAXAPRO to simplify their analysis process. Continued development of tools like TAXAPRO will revolutionize the way we analyze large WGS datasets and give anyone the ability to pursue metagenomics studies.


### Team
- Sofia Sehli (sofiasehli00@gmail.com)
- Casey Eddington (eddin022@umn.edu)
- Zainab El Ouafi (zainab.elouafi@gmail.com)
- Soumaya JBARA (soumaya.jbara@um5r.ac.ma)
- Kasambula Arthur Shem (arthurshem86@gmail.com)
- Islam El Jaddaoui (is.eljaddaoui@gmail.com)
- Ayorinde Afolayan (afolayanayorinde@gmail.com)
- Olaitan I. Awe (laitanawe@gmail.com)

### References
